pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -150,13 +156,67 @@ func (dv *openShiftClusterDynamicValidator) validateServicePrincipalProfile(ctx
 	return autorest.NewBearerAuthorizer(token), nil
 }
 
-func (dv *openShiftClusterDynamicValidator) validateVnetPermissions(ctx context.Context, oc *api.OpenShiftCluster, client authorization.PermissionsClient, code, typ string) error {
-	vnetID, _, err := subnet.Split(oc.Properties.MasterProfile.SubnetID)
+func (dv *openShiftClusterDynamicValidator) validateVnetPermissions(ctx context.Context, vnet mgmtnetwork.VirtualNetwork, oc *api.OpenShiftCluster, client authorization.PermissionsClient, code, typ string) error {
+	r, err := azure.ParseResourceID(*vnet.ID)
 	if err != nil {
 		return err
 	}
 
-	r, err := azure.ParseResourceID(vnetID)
+	timeoutCtx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+	defer cancel()
+
+	err = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+		permissions, err := client.ListForResource(ctx, r.ResourceGroup, r.Provider, r.ResourceType, "", r.ResourceName)
```

Comment:
I had a question regarding this function call.  I noticed that the `r.ResourceType` is being passed to the `ListForResource` in the `parentResourcePath` parameter position.  Was that intentional?  I would have thought that it would need to get passed in as the `resourceType` string.
```
	ListForResource(ctx context.Context, resourceGroupName string, resourceProviderNamespace string, parentResourcePath string, resourceType string, resourceName string) (permissions []mgmtauthorization.Permission, err error)
```

Reply:
Looking at the code, I think it might be the case that it doesn't actually matter which one is used, but that using the `resourceType` parameter is more correct, yes.

I was looking at vendor/github.com/Azure/azure-sdk-for-go/services/preview/authorization/mgmt/2018-09-01-preview/authorization/permissions.go , line 103.